### DISCUSSION TITLE
Abort old request if new request was made

### DISCUSSION THREAD

**Discussion by weeg** [livewire/livewire#2480] [3. Feature ideas]
It would be useful if the previous request would be aborted when a new request was send to the same component before the previous response arrived.

Now all responses of the requests are rendered. This can make the UI feel sluggish for slower responses.

For example:
1. When we have multiple checkboxes for filtering results. (These can be hard to check and uncheck because of the delayed response)
2. When we make typeahead search box and old highlights show up even if we have already entered new characters.

There was a similar discussion here: https://github.com/livewire/livewire/issues/724

**Comment by ikechukwukalu** [comment]
I was about to ask the same thing. Something similar to how `axios` handles theirs.

```
var ajaxRequest;
if (ajaxRequest)
   ajaxRequest.cancel(); // Aborts a previous requests

// creates a new token for upcomming ajax (overwrite the previous one)
ajaxRequest = axios.CancelToken.source();

// new request
axios.get();
```

**Comment by hamannjames** [comment]
I'm revitalizing this thread. There is a definite need here. @calebporzio if someone were to submit a PR that would expose an $abort or other type method to cancel handling a response and push up the queue would that be useful?
